docs/data/data-grid/group-pivot/RowGroupingGetGroupingCriteriaRows.js

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  gridColumnVisibilityModelSelector,
+  GridEvents,
+  useGridApiRef,
+} from '@mui/x-data-grid-pro';
+import { useMovieData } from '@mui/x-data-grid-generator';
+import Alert from '@mui/material/Alert';
+
+const INITIAL_GROUPING_COLUMN_MODEL = ['company', 'director'];
+
+const useKeepGroupingColumnsHidden = (apiRef, columns, initialModel, leafField) => {
+  const prevModel = React.useRef(initialModel);
+
+  React.useEffect(() => {
+    apiRef.current.subscribeEvent(GridEvents.rowGroupingModelChange, (newModel) => {
+      const columnVisibilityModel = {
+        ...gridColumnVisibilityModelSelector(apiRef),
+      };
+
+      newModel.forEach((field) => {
+        if (!prevModel.current.includes(field)) {
+          columnVisibilityModel[field] = false;
+        }
+      });
+      prevModel.current.forEach((field) => {
+        if (!newModel.includes(field)) {
+          columnVisibilityModel[field] = true;
+        }
+      });
+      apiRef.current.setColumnVisibilityModel(columnVisibilityModel);
+      prevModel.current = newModel;
+    });
+  }, [apiRef]);
+
+  return React.useMemo(
+    () =>
+      columns.map((colDef) =>
+        initialModel.includes(colDef.field) ||
+        (leafField && colDef.field === leafField)
+          ? { ...colDef, hide: true }
+          : colDef,
+      ),
+    [columns, initialModel, leafField],
+  );
+};
+
+export default function RowGroupingGetGroupingCriteriaRows() {
+  const data = useMovieData();
+  const apiRef = useGridApiRef();
+
+  const [lastGroupClickedChildren, setLastGroupClickedChildren] =
+    React.useState(null);
+
+  const columns = useKeepGroupingColumnsHidden(
+    apiRef,
+    data.columns,
+    INITIAL_GROUPING_COLUMN_MODEL,
+  );
+
+  const handleRowClick = React.useCallback(
+    (params) => {
+      // Only log groups
+      if (!apiRef.current.getRowNode(params.id)?.children) {
+        return;
+      }
+
+      const rowIds = apiRef.current.getGroupingCriteriaRows({
+        groupId: params.id,
+      });
+
+      const rowTitles = rowIds.map((rowId) => apiRef.current.getRow(rowId).title);
+
+      setLastGroupClickedChildren(rowTitles);
+    },
+    [apiRef],
+  );
+
+  return (
+    <div style={{ width: '100%' }}>
+      <div style={{ height: 400, width: '100%' }}>
+        <DataGridPro
+          {...data}
+          apiRef={apiRef}
+          columns={columns}
+          onRowClick={handleRowClick}
+          hideFooter
+          initialState={{
+            rowGrouping: {
+              model: INITIAL_GROUPING_COLUMN_MODEL,
+            },
+          }}
+          experimentalFeatures={{
+            rowGrouping: true,
+          }}
+        />
+      </div>
+      <Alert severity="info" style={{ marginBottom: 8 }}>
```

Comment:
To avoid inline style? It's usually discouraged for CSP 

```suggestion
      <Alert severity="info" sx={{ mb: 1 }}>
```

https://github.com/mui/material-ui/issues/19938#issuecomment-645893048

Reply:
We probably have a lot of other doc examples like this
Would you also replace the `div` with a `Box` to avoid inline style on them or not ?

Reply:
I'm not against the idea to replace div.